treesit-auto.el

```diff
@@ -408,13 +408,43 @@ how to modify the behavior of this function."
   ;; For this mode to keep a cached copy is dangerous as it will be a global
   ;; replacement and ignores all changes while this mode is active, so
   ;; don't think it is a valid option.
-  (setq-local major-mode-remap-alist (treesit-auto--build-major-mode-remap-alist)))
+  (when (treesit-auto--enabled-on-major-mode-p global-treesit-auto-modes mode)
+    (setq-local major-mode-remap-alist (treesit-auto--build-major-mode-remap-alist))))
 
 (defun treesit-auto--on ()
   "Turn `treesit-auto-mode' on."
   (setq-local treesit-language-source-alist
               (treesit-auto--build-treesit-source-alist))
   (treesit-auto--maybe-install-grammar))
 
+(defun treesit-auto--enabled-on-major-mode-p (predicate mode)
+  ;; Basically a copy of `easy-mmode--globalized-predicate-p', sans the
+  ;; (indirect) reliance on `major-mode' variable
+  (cond
+   ((eq predicate t)
+    t)
+   ((eq predicate nil)
+    nil)
+   ((listp predicate)
+    ;; Legacy support for (not a b c).
```

Comment:
What's the right way for us to use the `(not ...)`?  Should it be a first-class support or are we just making sure this is a backwards-compatible thing?  I couldn't quite figure out where I'd want to use `(not ...)` instead of simply removing the modes I don't like through the `M-x customize` menu or in the `:customize` section of `use-package`.